caikit_nlp/data_model/generation.py

```diff
@@ -71,3 +71,33 @@ class TuningConfig(DataObjectBase):
     # num_layers: int # Optional - The number of layers in the base transformer model
     #
     # encoder_hidden_size: int # Optional -  The hidden size of the prompt encoder.
+
+
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class DecodingParameters(DataObjectBase):
+    @dataobject
+    class ExponentialDecayLengthPenalty(DataObjectBase):
+        start_index: int
+        decay_factor: float
+
+    repetition_penalty: float
+    exponential_decay_length_penalty: ExponentialDecayLengthPenalty
+
+
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class SamplingParameters(DataObjectBase):
+
+    temperature: float
+    top_k: int
+    top_p: int
+    typical_p: float
+    seed: Optional[int]
```

Comment:
It might be a good idea to set some defaults here! TGIS defaults are [here](https://github.com/IBM/text-generation-inference/blob/main/router/src/lib.rs#L63).

Most of the time this doesn't matter, because 0 temperature (in the IBM fork) indicates greedy decoding, so `top_k`, `top_p`, `typical_p`, etc won't be used, as they're sampling only.

TGI doesn't use temperature 0 as a toggle though, so it would be also be nice in case those APIs are ever more unified - currently there is some small divergences with stuff like prompt IDs

Reply:
I think even if we set defaults on the DM, they won't propagate to proto, so the default here would be guided by the `.run` function themselves.

Reply:
Good point - my main concern with leaving it up to run is that it's easy for defaults to get out of sync if we have multiple modules relying on them. 

I guess an alternate is to either have a building for getting these objects with their default values that make sense, or to have consts be passed to the run function 🤔 is the intent with this type to have a parameter that is this DM object type, or to take primitives and build this object in the requests?